gax/src/main/java/com/google/api/gax/rpc/ClientContext.java

```diff
@@ -134,16 +140,15 @@ public static ClientContext create(StubSettings settings) throws IOException {
     final ScheduledExecutorService executor = executorProvider.getExecutor();
 
     Credentials credentials = settings.getCredentialsProvider().getCredentials();
+    if (settings.getQuotaProjectId() != null) {
+      credentials = new QuotaProjectIdHidingCredentials(credentials);
```

Comment:
This is interesting. I assume this is to make sure the quota project option isn't set twice (in call options and also channel options)? 

I would appreciate a comment here and/or a comment in the `QuotaProjectIdHidingCredentials` class explaining more about this and why it's necessary.

Reply:
> If a quotaProjectId is set, we also need to tell the Credentials to no longer send the x-goog-user-project header (it could be an incorrect value as the explicitly configured quotaProjectId setting should take precedence over the ones provided by the credentials).  
@chingor13 mentioned in previous comment.

Reply:
Yes, let's make sure to add a comment. We need to ensure that a custom set quota project id takes priority over one detected by credentials. If we don't, then credentials will also set the header and the backend will receive more than one (possibly conflicting) values.